modules/running-distributed-data-science-workloads-from-ds-pipeline.adoc

```diff
@@ -0,0 +1,133 @@
+           name="raytest",
+           # namespace must exist
+           namespace="pipeline-example",
+           num_workers=1,
+           head_cpus="500m",
+           min_memory=1,
+           max_memory=1,
+           num_gpus=0,
+           image="quay.io/project-codeflare/ray:latest-py39-cu118",
+           instascale=False,
+       )
+   )
+
+
+   print(cluster.status())
+   cluster.up()
+   cluster.wait_ready()
+   print(cluster.status())
+   print(cluster.details())
+
+
+   ray_dashboard_uri = cluster.cluster_dashboard_uri()
+   ray_cluster_uri = cluster.cluster_uri()
+   print(ray_dashboard_uri, ray_cluster_uri)
+
+
+   # before proceeding make sure the cluster exists and the uri is not empty
+   assert ray_cluster_uri, "Ray cluster needs to be started and set before proceeding"
+
+
+   ray.init(address=ray_cluster_uri)
+   print("Ray cluster is up and running: ", ray.is_initialized())
+
+
+   @ray.remote
+   def train_fn():
+       # complex training function
+       return 100
+
+
+   result = ray.get(train_fn.remote())
+   assert 100 == result
+   ray.shutdown()
+   cluster.down()
+   auth.logout()
+   return result
+
+
+@dsl.pipeline(
+   name="Ray Simple Example",
+   description="Ray Simple Example",
+)
+def ray_integration(openshift_server, openshift_token):
+   ray_op = components.create_component_from_func(
+       ray_fn,
+       base_image='registry.redhat.io/ubi8/python-39:latest',
+       packages_to_install=["codeflare-sdk"],
+   )
+   ray_op(openshift_server, openshift_token)
+
+
+if __name__ == '__main__':
+    from kfp_tekton.compiler import TektonCompiler
+    TektonCompiler().compile(ray_integration, 'compiled-example.yaml')
+
+----
+.. Install any required dependencies.
+In this example, install the `kfp` and `kfp-tekton` Python packages:
++
+[source,bash]
+----
+$ pip install kfp kfp-tekton
+----
+.. Compile the Python file (in this example, the `compile_example.py` file):
++
+[source,bash]
+----
+$ python compile_example.py
+----
+This command creates a YAML file (in this example, `compiled-example.yaml`), which you can import in the next step.
+. Import your data science pipeline as described in link:{rhoaidocshome}{default-format-url}/working_on_data_science_projects/working-with-data-science-pipelines_ds-pipelines#importing-a-data-science-pipeline_ds-pipelines[Importing a data science pipeline].
+. Schedule the pipeline run as described in link:{rhoaidocshome}{default-format-url}/working_on_data_science_projects/working-with-data-science-pipelines_ds-pipelines#managing_pipeline_runs[Scheduling a pipeline run].
+. When the pipeline run is complete, view the run details as described in link:{rhoaidocshome}{default-format-url}/working_on_data_science_projects/working-with-data-science-pipelines_ds-pipelines#viewing-the-details-of-a-pipeline-run_ds-pipelines[Viewing the details of a pipeline run].
+
+.Verification
+The YAML file is created and the pipeline run completes without errors.
```

Comment:
Would a record of this run be included in the "Triggered pipeline runs" interface?